enhancements/baremetal/ztp-metal3.md

```diff
@@ -0,0 +1,286 @@
+  automatically and handle coordination with Ironic and baremetal-operator.
+  The configuration will also supress automatic start-up of the assisted
+  agent.
+- [Ironic agent][ironic-agent] currently contains a custom deployment plugin
+  (*custom deploy method* in baremetal-operator terms, *custom deploy step* in
+  Ironic terms) that invokes `coreos-installer` for the image installation.
+  A new custom deploy method `install_assisted` will be added to start the
+  assisted agent at the right moment.
+- The assisted service will stop explicitly disabling *inspection* on
+  `BareMetalHost`s it manages, will start using a different process based
+  on the  new `install_assisted` deploy method and stop rebooting the machine
+  in the end of the process.
+
+### User Stories
+
+This change itself does not add new user stories, but rather enables existing
+features to work for ZTP:
+
+- As an operator, I want to ensure that all OpenShift nodes are deployed with
+  known firmware settings.
+- As an operator of hardware that does not support virtual media, I want to
+  deploy OpenShift nodes using iPXE network boot.
+- As an operator, I want OpenShift to ensure that the deployment ISO is
+  correctly disconnected on installation finish and cannot be accidentally
+  booted into.
+
+### API Extensions
+
+All required API extensions are already present.
+
+### Implementation Details/Notes/Constraints
+
+#### Image building
+
+ICC will follow different code paths depending on whether the `BareMetalHost`
+has the `infraenvs.agent-install.openshift.io` annotation linking it to an
+`InfraEnv` resource. In both cases it will generate an Ignition configuration
+to start and configure the Ironic agent. In the ZTP case,
+image-customization-controller will update the corresponding `InfraEnv` with
+the generated Ignition configuration (merging it with any user-provided
+configuration).
+
+To avoid putting the pull secret to the Ignition override of the `InfraEnv`
+resource, ICC will configure Ignition to use the pull secret already provided
+by the assisted service for its agent (`/root/.docker/config.json`).
+
+The same process will be applied to `initrd` images for (i)PXE booting. The
+Ignition configuration will be embedded into the resulting `initrd` by the
+assisted service. The root filesystem (required for CoreOS) will be provided
+by [machine-os-images][machine-os-images], similarly to the *standard* flow.
+
+#### Coordination between agents
+
+The first agent to start will be the Ironic agent since it is responsible
+for coordinating the whole flow and (indirectly through Ironic) updating the
+`BareMetalHost`'s status. *Inspection* and additional configuration (RAID etc)
+will be handled by the Ironic agent alone.
+
+BMAC will be able to stop synchronizing the inventory to `BareMetalHost`
+objects since this task will be handled by Ironic and baremetal-operator.
+
+Currently, the installation is initiated by BMAC (the assisted component
```

Comment:
BMAC doesn't initiate installation. It just boots a live ISO. Importantly, it is just one of many other ways a user could automate the booting of the live ISO.

Reply:
Well, it does initiate the installation from our (= BMO) perspective. I don't think we should add here a whole range of options AI provides.

Reply:
s/installation/provisioning/ would be more accurate